### DISCUSSION TITLE
Unable to run Puppeteer NodeJs script with symfony Process

### DISCUSSION THREAD

**Discussion by thgauthier** [symfony/symfony#50071] [Q&A]
Hi everyone !
I would like to run a NodeJs script with Puppeteer from a symfony controller.
Puppeteer has been installed with yarn like every package like webpack bootstrap...
When I run my test.js script from command line it run perfecftly.
But when I run it from a symfony controller with Process class, the test.js script start but fail with Puppeteer : 
`node_modules/puppeteer-core/lib/cjs/puppeteer/node/ProductLauncher.js:127`
Here my test.js script : 
![Capture d’écran du 2023-04-20 13-53-50](https://user-images.githubusercontent.com/95478233/233358341-3cdf8eb3-b153-4c4e-8712-e0fc3a20f7d0.png)
Here my symfony controller:

![Capture d’écran du 2023-04-20 13-56-01](https://user-images.githubusercontent.com/95478233/233358775-887a8f16-9843-4bc2-bab9-78d126faf75d.png)

Browser result : 

![Capture d’écran du 2023-04-20 13-59-55](https://user-images.githubusercontent.com/95478233/233359683-5d5351f3-0a82-47b5-8459-6bfdfcc9575b.png)
Any idea ?

**Comment by stefangr** [comment]
It seems this has nothing to do with symfony.

But you might want to surround the calls to puppeteer and the created browser instance (lines 10-17 in the test.js file) with `try...catch` and log the resulting error message.

Because I think the line after "output no 5" is part of one big nodejs error stacktrace.
